File: sol/1497/1497.py
class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        rem = [0] * k
        for x in arr:
            rem[x % k] += 1
        for x in range(k):
            if x == 0 or x * 2 == k:
                if rem[x] & 1:
                    return False
            elif rem[x] != rem[k - x]:
                return False
        return True

File: sol/1497/1497.cpp
struct Solution {
    bool canArrange(vector<int>& arr, int k) {
        vector<int> rem(k, 0);
        for (int x : arr){
            ++rem[((x % k) + k) % k];
        }
        for (int x = 0; x < k; ++x){
            if (x == 0 || x * 2 == k){
                if (rem[x] & 1){
                    return false;
                }
            } else if (rem[x] != rem[k - x]){
                return false;
            }
        }
        return true;
    }
};
